OIIInspector/ImageManager.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+from contextlib import suppress
+from OIIInspector.Exceptions import AddressAlreadyInUse, OIIInspectorError
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_max_tries = 5
+    _grpc_init_wait_time = 1000
+    _base_container_name = "OIIInspector_running_image"
+
+    def __init__(self):
+        self._container_platform = self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+
+    def start_image(self, image_address):
+        self._image_address = image_address
+        self._remove_old_image_if_exists()
+        if self._check_if_image_already_runs() is False:
+            self._pull_image()
+            self._port, self._rpc_proc = self._serve_index_registry()
+        else:  # pragma: no cover"
+            pass
+
+    def get_local_address_of_image(self):
+        return "localhost:{port}".format(port=self._port)
+
+    def close_image_manager(self):
+        self._stop_image_registry()
+        self._rpc_proc.kill()
+        self._remove_image_registry()
+
+    def _get_container_platform(self):
+        docker_terminal_result = run_cmd("command -v docker")
+        podman_terminal_result = run_cmd("command -v podman")
+        if "/podman" in podman_terminal_result:
+            return "podman"
+        elif "/docker" in docker_terminal_result:
+            return "docker"
+        else:
+            raise OIIInspectorError("Docker or Podman is needed to be installed on the platform")
+
+    def _remove_old_image_if_exists(self):
+        with suppress(Exception):
+            self._stop_image_registry(tolerate_err=True)
+        with suppress(RuntimeError):
+            self._remove_image_registry(tolerate_err=True)
+
+    def _stop_image_registry(self, tolerate_err=False):
+        run_cmd(f"{self._container_platform} stop {self._base_container_name}_{self._port}", tolerate_err=tolerate_err)
+
+    def _remove_image_registry(self, tolerate_err=False):
+        run_cmd(f"{self._container_platform} rm {self._base_container_name}_{self._port}", tolerate_err=tolerate_err)
+
+    def _check_if_image_already_runs(self):
+        return False
+
+    def _pull_image(self):
+        """
+        Pull needs to be done separately, because run can be interrupted with timer later (while running podman run).
```

Comment:
What does this mean?

Reply:
This was moved as coment to part of the code where this _pull_image method is used.